diesel/src/mysql/connection/bind.rs

```diff
@@ -40,6 +71,56 @@ impl Binds {
         &mut self.mysql_binds
     }
 
+    /// The caller of this function must immediately check for errors
+    /// after return
+    pub fn populate_dynamic_buffers(&mut self, stmt: *mut ffi::MYSQL_STMT) {
+        use std::cmp::min;
+
+        for (i, buffer) in self.data.iter_mut().enumerate() {
+            if self.errors.as_ref().unwrap()[i] != 0 {
+                let written_bytes = buffer.len();
+                let needed_capacity = self.lengths[i] as usize;
+                buffer.reserve(needed_capacity - written_bytes);
+                self.mysql_binds[i].buffer = buffer.as_mut_ptr() as *mut libc::c_void;
+                mem::swap(&mut self.lengths[i], &mut self.mysql_binds[i].buffer_length);
+                let load_result = unsafe {
+                    ffi::mysql_stmt_fetch_column(
+                        stmt,
+                        &mut self.mysql_binds[i],
+                        i as libc::c_uint,
+                        written_bytes as libc::c_ulong,
+                    )
+                };
+                if load_result != 0 {
+                    return;
+                }
+            }
+
+            let actual_bytes_in_buffer = min(
+                self.mysql_binds[i].buffer_length,
+                self.lengths[i],
+            );
+            unsafe { buffer.set_len(actual_bytes_in_buffer as usize) }
```

Comment:
Okay, I'm probably too tired to see this. Can _this_ `set_len` leak memory? Can you use `truncate` instead? Is this what you meant earlier? (The docs don't say much about `truncate` but it's source uses `drop_in_place` which may call destructors we don't want to run.)